includes/linux/LinuxKeyboard.h

```diff
@@ -68,26 +69,48 @@ namespace OIS
 			XPeekEvent(display, &e);
 			if(e.type == KeyPress && e.xkey.keycode == event.xkey.keycode && (e.xkey.time - event.xkey.time) < 2)
 			{
-				XNextEvent(display, &e);
+				XNextEvent(display, &e);
+				XFilterEvent(&event, None);
 				return true;
 			}
 
 			return false;
 		}
 
-		bool _injectKeyDown( KeySym key, int text );
-		bool _injectKeyUp( KeySym key );
-
-		//! 1:1 Conversion Map between X Key Events and OIS KeyCodes
-		typedef std::map<KeySym, KeyCode> XtoOIS_KeyMap;

```

Comment:
Okay, you are getting rid of the 1:1 "map" between key symbols and OIS::KeyCode. This is good, because this is a real pain.